File: types/event.ts
import * as anchor from "@project-serum/anchor"

export interface StartGameEvent {
    game: anchor.web3.PublicKey
}

export interface RevealQuestionEvent {
    game: anchor.web3.PublicKey
    question: anchor.web3.PublicKey
}

export interface RevealAnswerEvent {
    game: anchor.web3.PublicKey
    question: anchor.web3.PublicKey
}

File: programs/trivia/src/event.rs
use anchor_lang::prelude::*;

#[event]
pub struct EditGameEvent {
    pub game: Pubkey,
}

#[event]
pub struct RevealQuestionEvent {
    pub game: Pubkey,
    pub question: Pubkey,
}

#[event]
pub struct RevealAnswerEvent {
    pub game: Pubkey,
    pub question: Pubkey,
}
